.github/workflows/python-max-version.yml

```diff
@@ -0,0 +1,21 @@
+name: python-max-version
+on: [push]
+jobs:
+  checkout:
+    runs-on: ubuntu-latest
+    container: circleci/python:3.10.0
+    steps:
+      # This Docker file changes sets USER to circleci instead of using the default user, so we need to update file permissions for this image to work on GH Actions.
+      # See https://docs.github.com/actions/reference/virtual-environments-for-github-hosted-runners#docker-container-filesystem
+      - name: Setup file system permissions
+        run: sudo chmod -R 777 $GITHUB_WORKSPACE /github /__w/_temp
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
```

Comment:
```suggestion
        uses: actions/checkout@v3
        submodules: 'recursive'
```
This action, in addition to executing git commands, also authorizes Github, so we should also use it to handle submodules, otherwise, we can run into quote limits or other problems with unauthorized users.